src/portfolio.py

```diff
@@ -0,0 +1,299 @@
+
+############################################################################
+### GSCO 2024 - PORTFOLIO AND STRATEGY CLASSES
+############################################################################
+
+# --------------------------------------------------------------------------
+# Cyril Bachelard
+# This version:     13.06.2024
+# First version:    13.06.2024
+# --------------------------------------------------------------------------
+
+
+import pandas as pd
+import numpy as np
+from typing import Callable, Dict, List
+
+
+
+# --------------------------------------------------------------------------
+def floating_weights(X, w, start_date, end_date, rescale = True):
+
+    start_date = pd.to_datetime(start_date)
+    end_date = pd.to_datetime(end_date)
+    if start_date < X.index[0]:
+        raise ValueError('start_date must be contained in dataset')
+    if end_date > X.index[-1]:
+        raise ValueError('end_date must be contained in dataset')
+
+
+    w = pd.Series(w, index = w.keys())
+    if w.isna().any():
+        raise ValueError('weights (w) contain NaN which is not allowed.')
+    else:
+        w = w.to_frame().T
+    xnames = X.columns
+    wnames = w.columns
+
+    if not all(wnames.isin(xnames)):
+        raise ValueError('Not all assets in w are contained in X.')
+
+    X_tmp = X.loc[start_date:end_date, wnames].copy().fillna(0)
+    # short_positions = wnames[ w.iloc[0,:] < 0 ]
+    # if len(short_positions) > 0:
+    #     X_tmp[short_positions] = X_tmp[short_positions] * (-1)
+    xmat = 1 + X_tmp
+    # xmat.iloc[0] = w.dropna(how='all').fillna(0).abs()

```

Comment:
Do we need these commented out lines here?  If yes, plz add a comment that explains why they are important

Reply:
I added a TODO for this part.